renderer.js

```diff
@@ -980,47 +980,71 @@ class SDFGRenderer {
 
     // Updates buttons based on cursor mode
     update_toggle_buttons() {
-        if (this.mouse_mode == 'move') {
-            this.movemode_btn.innerHTML = '<i class="material-icons">done</i>';
-            this.movemode_btn.title = 'Exit object moving mode';
-        } else {
-            this.movemode_btn.innerHTML =
-                '<i class="material-icons">open_with</i>';
-            this.movemode_btn.title = 'Enter object moving mode';
-        }
-        if (this.mouse_mode == 'select') {
-            this.selectmode_btn.innerHTML =
-                '<i class="material-icons">done</i>';
-            this.selectmode_btn.title = 'Exit box select mode';
-        } else {
-            this.selectmode_btn.innerHTML =
-                '<i class="material-icons">border_style</i>';
-            this.selectmode_btn.title = 'Enter box select mode';
+        switch (this.mouse_mode) {
```

Comment:
Not sure a switch-case works here, since you need to disable the checkmark symbols for the other buttons (that's why I wrote it as if/else branches)

Reply:
Good catch, I overlooked that - I'll change it back to the if/else approach

Reply:
I decided to opt for the following approach; First clear out of all modes and 'reset' the interaction state, then select only the applicable one. 

Reasoning: With the if/else branches, it works fine for the two modes we currently have - however, if more modes are added, each with its own 'reset' actions, we run into the danger that whichever other mode's reset action follows a previous one, will make unintended changes. In other words, the order of if/else statements starts to matter, and you need to pay closer attention to what you're doing in each else branch.

What do you think?

Reply:
Yes, that works